stl/inc/vector

```diff
@@ -37,169 +37,207 @@ public:
 
     using _Tptr = typename _Myvec::pointer;
 
-    _Vector_const_iterator() noexcept : _Ptr() {}
+    _CONSTEXPR20_DYNALLOC _Vector_const_iterator() noexcept : _Ptr() {}
 
-    _Vector_const_iterator(_Tptr _Parg, const _Container_base* _Pvector) noexcept : _Ptr(_Parg) {
+    _CONSTEXPR20_DYNALLOC _Vector_const_iterator(_Tptr _Parg, const _Container_base* _Pvector) noexcept : _Ptr(_Parg) {
         this->_Adopt(_Pvector);
     }
 
-    _NODISCARD reference operator*() const noexcept {
+    _NODISCARD _CONSTEXPR20_DYNALLOC reference operator*() const noexcept {
 #if _ITERATOR_DEBUG_LEVEL != 0
-        const auto _Mycont = static_cast<const _Myvec*>(this->_Getcont());
-        _STL_VERIFY(_Ptr, "can't dereference value-initialized vector iterator");

```

Comment:
This is going to happen a lot, given that all other containers are getting the `constexpr` treatment

Should we adopt the `_STL_VERIFY` macro?